**CASO DE ESTUDIO**

Se desarrollará un modelo de Árbol de Decisión para predecir el riesgo de enfermedades cardíacas en pacientes, basándose en sus perfiles. El objetivo es clasificar a los individuos en dos estados: Sanos (0) o Enfermos (1), permitiendo así una identificación temprana de riesgos cardiovasculares.
Para que el modelo aprenda a identificar a un paciente en riesgo, utilizaremos las siguientes variables:

Age: Representa la edad cronológica del paciente.

Systolic: Indica la lectura de la presión de la sangre en las arterias cuando el corazón late. Si es muy alta, el corazón se esfuerza de más y puede dañarse.

Diastolic: Indica la lectura de la presión arterial cuando el corazón descansa entre latidos. Es clave para saber si las arterias están siempre bajo mucha tensión.

BMI: Relaciona el peso y la altura del paciente para determinar si existe sobrepeso u obesidad, factores que incrementan el esfuerzo del sistema cardiovascular.

Variable Objetivo:

Disease: Es el resultado final que el modelo busca adivinar.

0 (Sano): El paciente no presenta la enfermedad.

1 (Enfermo): El paciente sí presenta la enfermedad según los datos analizados.

0. Se procede con la carga de las librerías de trabajo.

In [23]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import confusion_matrix

1. Se procede con la carga de los datos del trabajo.

In [34]:
nxl='/content/2. HeartDisease.xlsx'
XDB=pd.read_excel(nxl,sheet_name=0)

XD=XDB.iloc[:,[1,2,3,4]]
yd=XDB.iloc[:,0]
display(XDB)

,Disease,Age,Systolic,Diastolic,BMI
0,0,44,112,111,17
1,1,55,128,90,27
2,1,47,131,94,26
3,0,31,151,104,17
4,1,65,148,117,17
...,...,...,...,...,...
175,0,43,119,76,25
176,1,63,107,113,40
177,0,47,116,61,27
178,0,70,110,88,24


2. Se procede con la implementación del módulo árbol.

In [35]:
mar=DecisionTreeClassifier(criterion='gini', max_depth=4)
mar.fit(XD,yd) #Aquí el modelo busca la relación entrada - salida.

#¿Y qué fue lo que hizo el modelo?

ydp=mar.predict(XD)   #Esto es lo que pronosticó el modelo.

#Se construye la matrix de confusion.
cm=confusion_matrix(yd,ydp)

display(cm)
VN=cm[0,0]; FP=cm[0,1]; FN=cm[1,0]; VP=cm[1,1]

#Métricas de desempeño

Ex= (VP+VN)/len(XD)         #1. EXACTITUD: COMPORTAMIENTO GENERAL
print("La exactitud es:", Ex)

Sen=VP/(VP+FN)              #2. SENSIBILIDAD; COMO SE COMPORTA PRONOSTICANDO PREAPROBADOS.
print("La sensibilidad es:", Sen)

Spe=VN/(VN+FP)              #3. ESPECIFICIDAD: COMO SE COMPORTA FRENTE A LOS NEGATIVOS.
print("La especificidad es:", Spe)

Pre=VP/(VP+FP)              #4. PRECISION: COMO SE COMPORTA PRONOSTICANDO PREAPROBADOS
print("La precisión es:",Pre)

PreNeg=VN/(VN+FN)           #5. PRECISION NEGATIVA: COMO SE COMPORTA PRONOSTICANDO NEGATIVOS
print("La precisión negativa es:", PreNeg)

array([[63, 10],
       [20, 87]])

La exactitud es: 0.8333333333333334
La sensibilidad es: 0.8130841121495327
La especificidad es: 0.863013698630137
La precisión es: 0.8969072164948454
La precisión negativa es: 0.7590361445783133


**ANÁLISIS DE RESULTADOS**
De la base de datos se puede observar un total de 73 personas que poseen la categoría de Sanos (0), mientras que para la categoría de Enfermos (1) la matriz cuenta con un total de 107 personas. De los 73 datos de personas sanas, el modelo pronosticó correctamente un total de 63 datos (Verdaderos Negativos), mientras que de los 107 enfermos, el modelo pronosticó correctamente un total de 87 datos (Verdaderos Positivos).

El modelo logra identificar en un 86.30% a las personas sanas (63/73), mientras que logró identificar en un 81.31% a las personas enfermas (87/107). Con respecto a las métricas, podemos observar que el modelo logró una Exactitud de 83.33%, lo que indica el buen comportamiento general del modelo frente a la clasificación en las dos categorías de salud. Se destacan la especificidad y la precisión, las cuales lograron valores por encima del 86% y 89% respectivamente, lo que supera ampliamente el límite inferior definido para este tipo de modelos de clasificación, el cual se ubica normalmente en el 75%.

De acuerdo con el árbol de decisión, se destaca la exploración del lado derecho (basado en niveles altos de BMI y Presión Sistólica) donde se encuentran nodos con alta pureza. Se identifica una regla de decisión extrema: SI el paciente posee un BMI > 28.39, una Edad > 45.87 y una Presión Sistólica > 125.58, tendrá una probabilidad de enfermedad del 100% (Nodo puro con Gini = 0). Esto demuestra que el modelo diferencia con gran precisión a los pacientes con perfiles de alto riesgo coronario.

A pesar de la existencia de falsos negativos, se destacan las reglas de negocio para la prevención: el modelo es altamente confiable al emitir diagnósticos positivos, con una Precisión del 89.69%. Esto significa que cuando el sistema clasifica a un sujeto como enfermo, la probabilidad de acierto es casi total, permitiendo priorizar la atención médica en esos casos críticos identificados en el pronóstico final.

3.Despliegue del Árbol de Decisión

In [36]:
from sklearn.tree import export_graphviz           #Exporta los datos a un gráfico
from pydotplus import graph_from_dot_data          #Es un graficador

vs=['Age', 'Systolic', 'Diastolic', 'BMI']         #Títulos del árbol
dot_data=export_graphviz(mar,feature_names=vs)     #Exportar de números a gráfico en pdf
graph=graph_from_dot_data(dot_data)                #Hacemos el gráfico
graph.write_png('Arbol.png')

True

4. Se realiza el pronóstico de 8 perfiles.

In [39]:
# 1. Carga de la Hoja 2 (Datos Pronóstico)
# Asegúrate de que el nombre del archivo sea correcto
nxl = '/content/2. HeartDisease.xlsx'
df_pronostico = pd.read_excel(nxl, sheet_name=1) # El índice 1 es la Hoja 2

# 2. Selección de las variables de soporte para los 8 sujetos
# Usamos las mismas columnas: Age, Systolic, Diastolic, BMI
X_pronostico = df_pronostico.iloc[:, [1, 2, 3, 4]]

# 3. Aplicación del modelo entrenado (mar)
# El modelo 'mar' ya debe estar entrenado con la Hoja 1
predicciones_hoja2 = mar.predict(X_pronostico)

# 4. Presentación de resultados
df_pronostico['Resultado_Enfermedad'] = predicciones_hoja2

print("--- RESULTADOS DEL PRONÓSTICO PARA LA HOJA 2 ---")
display(df_pronostico[['Age', 'Systolic', 'Diastolic', 'BMI', 'Resultado_Enfermedad']])

--- RESULTADOS DEL PRONÓSTICO PARA LA HOJA 2 ---


,Age,Systolic,Diastolic,BMI,Resultado_Enfermedad
0,33,120,64,33,0
1,55,115,81,32,0
2,51,130,70,20,0
3,65,101,63,18,1
4,62,109,110,38,1
5,60,130,69,35,1
6,61,124,95,36,1
7,53,150,81,41,1


**ANÁLISIS DE RESULTADOS**
Tras aplicar el modelo a la Hoja 2, se clasificó a 3 sujetos como Sanos (0) y a 5 como Enfermos (1). El análisis destaca al Sujeto 8 como el perfil de mayor riesgo, ya que cumple la regla del nodo puro al combinar un BMI de 41, una edad de 53 años y una presión sistólica de 150, situándose en la rama de probabilidad de enfermedad del 100%. Por el contrario, sujetos como el 1 y 2 fueron clasificados como sanos debido a que su juventud (33 años) o sus niveles controlados de presión arterial (115 mmHg) los mantienen en las ramas de seguridad del árbol, demostrando que el modelo pondera estos factores como protectores frente al riesgo cardiovascular.

En conclusión, el pronóstico sobre los 8 perfiles demuestra la consistencia del modelo, ya que las clasificaciones guardan una relación lógica con los factores de riesgo médicos. Los resultados confirman que el árbol segmenta eficazmente a los pacientes, identificando que la combinación crítica de obesidad (BMI alto) y tensión arterial elevada, especialmente en personas mayores de 45 años, son los detonantes principales de la enfermedad. Este análisis valida la utilidad de la herramienta para priorizar la atención médica en los casos de mayor vulnerabilidad identificados en la Hoja 2.